### DISCUSSION TITLE
Why are API tokens not default active?

### DISCUSSION THREAD

**Discussion by vince1995** [strapi/strapi#6262] [Legacy Topics]
Hi,

I read your [guide for API tokens](https://strapi.io/documentation/3.0.0-beta.x/guides/api-token.html) and took a look at your code. I've noticed that you could already support API token because [this file](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/services/Jwt.js) already accepts the token in the URL query. So I've replaced the code [from here](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/config/policies/permissions.js)

```
module.exports = async (ctx, next) => {
  let role;

  if (ctx.request && ctx.request.header && ctx.request.header.authorization) {
    try {
      const { id, isAdmin = false } = await strapi.plugins[
        'users-permissions'
      ].services.jwt.getToken(ctx);

      if (id === undefined) {
        throw new Error('Invalid token: Token did not contain required fields');
      }
```

based on the API token guide with this here:

```
module.exports = async (ctx, next) => {
  let role;

  // Check if there's a token in the request.
  const hasTokenInQuery = ctx.request && ctx.request.query && ctx.request.query.token;

  // Accept the token.
  if ((ctx.request && ctx.request.header && ctx.request.header.authorization) || hasTokenInQuery) {
    try {
      const { id, isAdmin = false } = await strapi.plugins[
        'users-permissions'
      ].services.jwt.getToken(ctx);
      
      // Delete the token from the request.
      if(hasTokenInQuery){
        delete ctx.request.query.token;
      }

      if (id === undefined) {
        throw new Error('Invalid token: Token did not contain required fields');
      }
```

And it works. But what's the problem why it's not in the code already? 🤔

**Comment by lauriejim** [comment]
Hello! Here is a simple implementation of the feature.
Developing a new feature require time for design UI / development, insert it in the roadmap and depending of available resources and priorities.
We are focus on the stable version that should be released really soon.
Here is the feature in the roadmap - [roadmap](https://portal.productboard.com/strapi/1-public-roadmap/c/40-api-access-token-with-permissions)
